packages/wrangler/src/__tests__/tail.test.ts

```diff
@@ -16,195 +17,221 @@ describe("tail", () => {
-      expectedWebsocketMessage
-    );
-  });
+  describe("printing", () => {
+    it("logs request messages in JSON format", async () => {
+      await runWrangler("tail test-worker --format json");
 
-  /* Printing */
+      const event = generateMockRequestEvent();
+      const message = generateMockEventMessage({ event });
+      const serializedMessage = serialize(message);
 
-  it("logs incoming messages in JSON format", async () => {
-    await runWrangler("tail test-worker --format json");
+      api.ws.send(serializedMessage);
+      expect(std.out).toMatch(deserializeToJson(serializedMessage));
+    });
 
-    /* request event */
-    const requestEvent = generateMockRequestEvent();
-    const requestMessage = generateMockEventMessage({ event: requestEvent });
-    const serializedRequestMessage = serialize(requestMessage);
+    it("logs scheduled messages in JSON format", async () => {
+      await runWrangler("tail test-worker --format json");
 
-    api.ws.send(serializedRequestMessage);
-    expect(std.out).toMatch(deserializeToJson(serializedRequestMessage));
+      const event = generateMockScheduledEvent();
+      const message = generateMockEventMessage({ event });
+      const serializedMessage = serialize(message);
 
-    /* scheduled event */
-    const scheduledEvent = generateMockScheduledEvent();
-    const scheduledMessage = generateMockEventMessage({
-      event: scheduledEvent,
+      api.ws.send(serializedMessage);
+      expect(std.out).toMatch(deserializeToJson(serializedMessage));
     });
-    const serializedEventMessage = serialize(scheduledMessage);
 
-    api.ws.send(serializedEventMessage);
-    expect(std.out).toMatch(deserializeToJson(serializedEventMessage));
-  });
+    it("logs request messages in pretty format", async () => {
+      await runWrangler("tail test-worker --format pretty");
 
-  it("logs messages in pretty-printing format", async () => {
-    await runWrangler("tail test-worker --format pretty");
+      const event = generateMockRequestEvent();
+      const message = generateMockEventMessage({ event });
+      const serializedMessage = serialize(message);
 
-    /* request event */
-    const requestEvent = generateMockRequestEvent();
-    const requestMessage = generateMockEventMessage({ event: requestEvent });
-    const serializedRequestMessage = serialize(requestMessage);
+      const method = event.request.method.toUpperCase();
+      const url = event.request.url;
+      const outcome = prettifyOutcome(message.outcome);
+      const datetime = new Date(message.eventTimestamp).toLocaleString();
 
-    api.ws.send(serializedRequestMessage);
-    expect(std.out).not.toMatch(deserializeToJson(serializedRequestMessage));
-
-    /* scheduled event */
-    const scheduledEvent = generateMockScheduledEvent();
-    const scheduledMessage = generateMockEventMessage({
-      event: scheduledEvent,
+      api.ws.send(serializedMessage);
+      expect(std.out).toMatchInlineSnapshot(`
```

Comment:
Don't parameterise inline snapshots; if this implementation breaks, then you won't catch the error because this test will still pass. We shouldn't be testing the internals like this, I don't think you should be importing `prettifyOutcome` here. Let the snapshot capture the whole output?

Reply:
Ok